### PR TITLE
feat(api): update news and media mentions, add events entity with complete CRUD functionality

### PR DESCRIPTION
**Pull request by Fedorieieva**
## Description

This PR introduces a significant refactoring of the content management system to ensure consistency across different modules and localization support. 

Key changes include:
#### 1. Schema Unification & Alignment
Previously, there were inconsistencies between `News` and `MediaMentions`. This PR aligns them under a standardized `BaseContent` structure. 
- **Consistency Fixes**: 
    - **Localization**: `MediaMentions` previously used flat `String` types for `title` and `description`. They have been updated to `LocalizedString` to match `News` and `Events`.
    - **SEO Fields**: Added mandatory `adminTitle`, `keywords` (localized), and `allowIndexation` (localized boolean) to all entities to ensure uniform metadata handling.
    - **Image Metadata**: Unified `LocalizedImage` schema to include optional `width` and `height` fields for frontend optimization.

#### 2. Advanced API Logic (GraphQL)
- **Complex Sorting**: Filters now accept a `sort` array, allowing users to sort by multiple fields in a single query.
- **Enhanced Filtering**: Updated `NewsFiltersInput`, `MediaMentionsFiltersInput`, and the new `EventFiltersInput` to support the updated status enums and slug-based lookups.

#### 3. Events Module (New)
- Developed the **Events entity from scratch**, as no prior implementation existed in branch develop.
- Created the Mongoose model, TypeScript entities, and a full set of GraphQL operations (Queries: `eventById`, `eventBySlug`, `paginatedEvents`; Mutations: `createEvent`, `updateEvent`, `deleteEvent`, `incrementEventViews`).

#### 4. Infrastructure & Maintenance
- **Data Migration**: Executed a comprehensive migration script to:
    - Transform legacy flat strings in `MediaMentions` into localized objects.
    - Convert all date fields to ISO strings for consistency.
    - Populate missing mandatory SEO fields with default values for existing records.

#### 5. Testing & Quality Assurance
- **Comprehensive Test Updates**: Existing tests for `News` and `MediaMentions` were updated to accommodate the new localized structures and mandatory SEO fields.
- **New Test Coverage**: Developed a full suite of tests for the newly created `Events` module.
- **Validation Logic**: Added tests to ensure the multi-criteria sorting and advanced filtering logic correctly interface with the database layer.

#### 6. Code Reusability & Generic Helpers
To eliminate code duplication  several core logic blocks were extracted from the existing News and Media Mentions resolvers into a centralized `helpers` file. These methods have been rewritten as generics to support all content types, including the new Events module. All helpers are covered by unit tests to ensure robustness and correct edge-case handling.

- **Refactored Resolver Helpers**:
    - **`mapFilters`**: A generic utility that standardizes the transformation of GraphQL `FiltersInput` into repository-ready filter objects. It now supports the new complex multi-criteria sorting logic across all modules.
    - **`extractTitleForSlug`**: Centralized the logic for to ensure consistent slug generation.
    - **`processSlugUpdate`**: Generic async helper for slug regeneration across any entity that implements a `findBySlug` method.

In addition to resolver logic, core data access patterns were extracted from newsRepository and mediaMentionRepository into a shared helpers file to ensure nified transformation standards and leaves repositories focused on entity-specific logic. All extracted repository helpers are fully covered by tests.

- **Standardized Repository Helpers**:
    - **`createToEntity`**: A generic mapper that automates the conversion of MongoDB documents (with `_id`, `createdAt`, `updatedAt`) into domain entities. This replaces manual mapping previously found in each repository.
    - **`buildBaseQuery`**: A unified query builder that handles common filtering parameters like `status` and `slug`. It replaces the redundant `buildNewsQuery` and `mediaMentionQueryBuilder` functions.
    - **`getBaseSort`**: Centralized logic for handling the new complex sorting structure.

## Type of change

- [x] New feature
- [ ] Bug fix
- [ ] Documentation update
- [x] Refactoring 
- [ ] CI/CD improvement
- [ ] Other: 

## ScreenShots after modifications
image
image
image

Closes: #[380](https://github.com/Liatoshynsky-Foundation/lf-admin/issues/380)

### CODE REVIEW

File: src/interfaces/graphql/resolvers/events/EventsMutation.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      expect(mockRepo.update).toHaveBeenCalled();
+    });
+
+    it('should default to Draft status if no status is provided', async () => {
+      const input = createMockInput();
+      const { status: _status, ...inputWithoutStatus } = input;
+
+      (mockRepo.create as jest.Mock).mockResolvedValue(createMockEntity({ status: EventStatus.Draft }));
+
+      await EventsMutation.createEvent({}, { input: inputWithoutStatus as any }, adminContext);
```
Comment (DenisGordProgrammer) [inline src/interfaces/graphql/resolvers/events/EventsMutation.test.ts]: Warning!  Unexpected any. Specify a different type :D

File: src/interfaces/graphql/resolvers/news/NewsQuery.test.ts [hunk 2]
```diff
@@ -1,25 +1,98 @@
...
-    await expect(NewsQuery.allNews({}, { filters: {} } as any, { admin: false } as any)).rejects.toThrow();
+    const invalidContext = { admin: false } as unknown as GraphQLContext;
+    const args = { filters: {} as NewsFiltersInput };
+
+    await expect(NewsQuery.allNews({}, args, invalidContext)).rejects.toThrow();
+  });
+
+  describe('newsCount', () => {
+    it('should call count without status if not provided', async () => {
+      await NewsQuery.newsCount({}, { status: NewsStatus.Archived as any }, context);
```
Comment (DenisGordProgrammer) [inline src/interfaces/graphql/resolvers/news/NewsQuery.test.ts]: Warning!  Unexpected any. Specify a different type :D

File: src/interfaces/graphql/resolvers/helpers.test.ts [hunk 3]
```diff
@@ -1,22 +1,144 @@
...
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should update slug property in updateData', async () => {
+      mockRepo.findBySlug.mockResolvedValue(null);
+      const updateData: { slug?: string } = {};
+
+      await processSlugUpdate('123', 'Новий заголовок', mockRepo as any, updateData);
```
Comment (DenisGordProgrammer) [inline src/interfaces/graphql/resolvers/helpers.test.ts]: Warning!  Unexpected any. Specify a different type :D

File: src/interfaces/graphql/resolvers/helpers.test.ts [hunk 4]
```diff
@@ -1,22 +1,144 @@
...
+
+      expect(updateData.slug).toBeDefined();
+      expect(typeof updateData.slug).toBe('string');
+      expect(mockRepo.findBySlug).toHaveBeenCalled();
+    });
+
+    it('should not update slug if title is missing', async () => {
+      const updateData: { slug?: string } = {};
+      await processSlugUpdate('123', null, mockRepo as any, updateData);
```
Comment (DenisGordProgrammer) [inline src/interfaces/graphql/resolvers/helpers.test.ts]: Warning!  Unexpected any. Specify a different type :D

**Review by DenisGordProgrammer — CHANGES_REQUESTED** [review]
Image

**Comment by MaksFullJs** [comment]
You should definitely fix sonar qube issues

File: app/types/graphql/queries/events.graphql
Comment (vlad-khrychov) [inline app/types/graphql/queries/events.graphql:1]: I’d suggest extracting the shared selection set into a GraphQL fragment.
`EventById` and `EventBySlug` currently duplicate the same fields, which can become hard to maintain if the schema evolves.

Look at https://www.apollographql.com/docs/react/data/fragments

File: migrations/1773128824997-unify-news-media-content-structures-add-seo-fields-and-events-schema.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+          },
+          'coverImage.caption': { uk: '', en: '' }
+        }
+      }
+    );
+  }
+}
+
+async function migrateNews(db: Db, now: string): Promise<void> {
+  const news = await db.collection('news').find({}).toArray();
```
Comment (vlad-khrychov) [inline migrations/1773128824997-unify-news-media-content-structures-add-seo-fields-and-events-schema.ts]: We may want to avoid `toArray()` here and iterate over a cursor instead, since loading the whole collection into memory can become risky for larger datasets.

I received a suggestion like this from gpt chat
```
const cursor = db.collection('news').find({});

for await (const doc of cursor) {
  await db.collection('news').updateOne(
    { _id: doc._id },
    {
      $set: {
        adminTitle: doc.title?.uk || 'Untitled News',
        keywords: { uk: '', en: '' },
        allowIndexation: { uk: true, en: true },
        newsDate: doc.newsDate ? new Date(doc.newsDate).toISOString() : null,
        publishedAt: doc.publishedAt ? new Date(doc.publishedAt).toISOString() : null,
        createdAt: doc.createdAt ? new Date(doc.createdAt).toISOString() : now,
        updatedAt: now
      }
    }
  );
}
```

File: migrations/1773128824997-unify-news-media-content-structures-add-seo-fields-and-events-schema.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+          $unset: {
+            adminTitle: '',
+            keywords: '',
+            allowIndexation: '',
+            'coverImage.caption': ''
+          }
+        }
+      );
+    }
+  }
```
Comment (vlad-khrychov) [inline migrations/1773128824997-unify-news-media-content-structures-add-seo-fields-and-events-schema.ts:135]: `down()` does not look fully symmetrical to `up()`. We add several fields during migration (`createdAt`, `updatedAt`, `publishedAt`, `newsDate`, `meta` etc.), but only some of them are reverted. This could make rollback incomplete.

File: src/infrastructure/repositories/eventRepository/eventRepository.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    content: doc.content,
+    coverImage: doc.coverImage,
+    status: doc.status,
+    meta: doc.meta,
+    publishedAt: doc.publishedAt ?? undefined,
+  });
+
+export const EventsRepository = ({ EventModel }: EventRepoDeps): IEventsRepository => {
+  const baseRepo = createBaseRepository<EventsEntity, DbEvent, EventFilters>({
+    model: EventModel as unknown as Model<DbEvent>,
```
Comment (vlad-khrychov) [inline src/infrastructure/repositories/eventRepository/eventRepository.ts]: The `as unknown as Model<DbEvent>` cast looks like a type-safety escape hatch. It may be worth aligning the model/document typings so we do not need a double cast here.

File: src/infrastructure/repositories/eventRepository/eventRepository.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      const eventData = {
+        ...input,
+        meta: { views: input.meta?.views ?? 0 }
+      };
+      const newEvent = await new EventModel(eventData).save();
+      return toEntity(newEvent.toObject() as unknown as DbEvent);
+    },
+    incrementViews: async (id: string): Promise<EventsEntity | null> => {
+      await dbConnect();
+      if (!/^[0-9a-fA-F]{24}$/.test(id)) return null;
```
Comment (vlad-khrychov) [inline src/infrastructure/repositories/eventRepository/eventRepository.ts]: Instead of validating ObjectId via regex, it may be safer/more idiomatic to use Mongoose’s built-in `ObjectId.isValid()` helper.

File: src/infrastructure/repositories/eventRepository/eventRepository.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    slug: string;
+    coverImage: EventsEntity['coverImage'];
+    status: EventStatus;
+    meta: EventsEntity['meta'];
+    createdAt: string;
+    updatedAt: string;
+};
+
+type EventRepoDeps = Readonly<{
+    EventModel: Model<EventsEntity>;
```
Comment (vlad-khrychov) [inline src/infrastructure/repositories/eventRepository/eventRepository.ts]: Try this 
```suggestion
    EventModel: Model<DbEvent>;
```

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
I haven’t gone through the entire PR in detail yet, but I have a few suggestions.
Please tag me in Discord when there’s an updated version and I’ll take another pass at the review.

**Review by vlad-khrychov — APPROVED** [review]

File: src/domain/repositories/baseRepository.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    createdAt: Date | string;
+    updatedAt: Date | string;
+};
+
+export interface IBaseRepository<TEntity extends BaseEntity, TFilters extends FiltersInput = FiltersInput> {
+    findById(id: string): Promise<TEntity | null>;
+    findBySlug(slug: string): Promise<TEntity | null>;
+    findAll(filters?: TFilters): Promise<TEntity[]>;
+    update(id: string, input: Partial<Omit<TEntity, keyof BaseEntity>>): Promise<TEntity | null>;
+    delete(id: string): Promise<boolean>;
```
Comment (Jevgan) [inline src/domain/repositories/baseRepository.ts:30]: I thought it would be better if delete mutation returned `id` of deleted publication instead of boolean to make code audit more understandable.

**Review by Jevgan — APPROVED** [review]

File: src/infrastructure/repositories/mediaMentionRepository/mediaMentionRepository.test.ts
Comment (navimov) [inline src/infrastructure/repositories/mediaMentionRepository/mediaMentionRepository.test.ts:1]: Small suggestion: could we rename this file to `mediaMentionRepository.test.ts` to better match its purpose?

File: src/infrastructure/repositories/mediaMentionRepository/mediaMentionRepository.ts
Comment (navimov) [inline src/infrastructure/repositories/mediaMentionRepository/mediaMentionRepository.ts:1]: Small suggestion: could we rename this file to `mediaMentionRepository.ts` to better match its purpose?

File: src/interfaces/graphql/resolvers/media-mentions/mediaMentionMutation.test.ts
Comment (navimov) [inline src/interfaces/graphql/resolvers/media-mentions/mediaMentionMutation.test.ts:1]: Small suggestion: could we rename this file to `mediaMentionMutation.test.ts` to better match its purpose?

File: src/interfaces/graphql/resolvers/media-mentions/mediaMentionMutation.ts
Comment (navimov) [inline src/interfaces/graphql/resolvers/media-mentions/mediaMentionMutation.ts:1]: Small suggestion: could we rename this file to `mediaMentionMutation.ts` to better match its purpose?

File: src/interfaces/graphql/resolvers/media-mentions/mediaMentionQuery.test.ts
Comment (navimov) [inline src/interfaces/graphql/resolvers/media-mentions/mediaMentionQuery.test.ts:1]: Small suggestion: could we rename this file to `mediaMentionQuery.test.ts` to better match its purpose?

File: src/interfaces/graphql/resolvers/media-mentions/mediaMentionQuery.ts
Comment (navimov) [inline src/interfaces/graphql/resolvers/media-mentions/mediaMentionQuery.ts:1]: Small suggestion: could we rename this file to `mediaMentionQuery.ts` to better match its purpose?

**Review by navimov — APPROVED** [review]

**Review by DenisGordProgrammer — APPROVED** [review]

**Review by MaksFullJs — APPROVED** [review]

**Review by kolibri753 — APPROVED** [review]

**Review by Jevgan — APPROVED** [review]
